### PR TITLE
[FLINK-32653][docs] Add doc for catalog store

### PR DESCRIPTION
**Pull request by hackergin**
## What is the purpose of the change

*Add doc for catalog store*

## Brief change log

* Add doc for catalog store* 

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? ( docs)

### CODE REVIEW

File: docs/content/docs/dev/table/catalogs.md [hunk 1]
```diff
@@ -738,3 +738,192 @@ Flink SQL> show tables;
 ```
 {{< /tab >}}
 {{< /tabs >}}
+
+## Catalog Store
+
+Catalog Store is used to store the configuration of catalogs. When using Catalog Store, the configurations
+of catalogs created in the session will be persisted in the corresponding external system of Catalog Store.
```
Comment (FangYongs) [inline docs/content/docs/dev/table/catalogs.md:745]: What's `session` here? Besides sql-gateway, there's no session in flink. The session here is a little strange

File: docs/content/docs/dev/table/catalogs.md [hunk 2]
```diff
@@ -738,3 +738,192 @@ Flink SQL> show tables;
...
+final EnvironmentSettings settings =
+        EnvironmentSettings.newInstance().inBatchMode()
+        .withCatalogStore(catalogStore)
+        .build();
+
+final TableEnvironment tableEnv = TableEnvironment.create(settings);
+
+// Register a catalog store using configuration.
+        
+// Set up configuration
```
Comment (FangYongs) [inline docs/content/docs/dev/table/catalogs.md:769]: Can we put them into different code blocks? It's a bit strange to put them together

File: docs/content/docs/dev/table/catalogs.md [hunk 3]
```diff
@@ -738,3 +738,192 @@ Flink SQL> show tables;
...
+```yaml
+table.catalog-store.kind: file
+table.catalog-store.file.path: /path/to/catalog/store/
+```
+{{< /tab >}}
+{{< /tabs >}}
+
+### Catalog Store Type
+Flink has two built-in Catalog Stores, namely GenericInMemoryCatalogStore and FileCatalogStore.
+Users can also customize their own Catalog Store.
```
Comment (FangYongs) [inline docs/content/docs/dev/table/catalogs.md:796]: Give an example for {catalog name}.yaml in the file path

**Comment by hackergin** [comment]
@FangYongs Thanks for your review, I update the code.

File: docs/content/docs/dev/table/catalogs.md (hunk 1 above)
Comment (hackergin) [reply docs/content/docs/dev/table/catalogs.md:814]: I think `session` represent a execution environment, not only in sql-gateway.  Such as session is used in bellow doc. 

https://github.com/apache/flink/blob/fc2b5d8f53a41695117f6eaf4c798cc183cf1e36/docs/content/docs/dev/table/catalogs.md?plain=1#L45

https://github.com/apache/flink/blob/fc2b5d8f53a41695117f6eaf4c798cc183cf1e36/docs/content/docs/dev/table/timezone.md?plain=1#L29

File: docs/content/docs/dev/table/catalogs.md [hunk 4]
```diff
@@ -807,3 +807,192 @@ the gateway, or you can also use `SET` to specify the listener for ddl, for exam
...
+    <tr>
+      <td><h5>kind</h5></td>
+      <td>Specify the Catalog Store type to be used, which should be 'generic_in_memory'</td>
+    </tr>
+    </tbody>
+</table>
+
+#### FileCatalogStore
+FileCatalogStore can save the user's Catalog configuration to a file. Currently, it only supports
+local files. To use FileCatalogStore, you need to specify the directory where the Catalog configuration
```
Comment (FangYongs) [inline docs/content/docs/dev/table/catalogs.md:883]: Currently FLINK-32660 has been merged and catalog store supports external file systems, I think we should update the doc

**Comment by FangYongs** [comment]
@hackergin Just one minor comment

**Comment by hackergin** [comment]
@FangYongs Thanks for reviewing the pr.

**Comment by FangYongs** [comment]
Thanks @hackergin , LGTM

**Pull request merged**